tools/lambda-promtail/lambda-promtail/main_test.go

```diff
@@ -64,3 +64,33 @@ func TestLambdaPromtail_TestDropLabels(t *testing.T) {
 	require.NotContains(t, modifiedLabels, model.LabelName("A1"))
 	require.Contains(t, modifiedLabels, model.LabelName("B2"))
 }
+
+func TestLambdaPromtail_ExtraHeadersValid(t *testing.T) {
+	extraHeaders, err := parseExtraHeaders("X-Custom-Header,This!sATota\\yCu$t0mHe4der,My-Server_WantsThis,What_ever could go here?,Expected4Entry,yLKc+QSB5VF/Gp3VPN7oOxa98yxWMxeHOAo+CW6trow=")
+	require.Nil(t, err)
+	require.Len(t, extraHeaders, 3)
+	require.Equal(t, extraHeaders["X-Custom-Header"], "This!sATota\\yCu$t0mHe4der")
+	require.Equal(t, extraHeaders["My-Server_WantsThis"], "What_ever could go here?")
+	require.Equal(t, extraHeaders["Expected4Entry"], "yLKc+QSB5VF/Gp3VPN7oOxa98yxWMxeHOAo+CW6trow=")
+}
+
+func TestLambdaPromtail_ExtraHeadersInvalidHeaderKey(t *testing.T) {
+	extraHeaders, err := parseExtraHeaders("Th.s_Shou|d-Fa!l,a")
+	require.Nil(t, extraHeaders)
+	require.ErrorContains(t, err, "HTTP header key is invalid:")
+	extraHeaders, err = parseExtraHeaders("Also Not Valid ,b")
+	require.Nil(t, extraHeaders)
+	require.ErrorContains(t, err, "HTTP header key is invalid:")
+}
+
+func TestLambdaPromtail_ExtraHeadersMissingValue(t *testing.T) {
+	extraHeaders, err := parseExtraHeaders("A,a,B,b,C,c,D")
+	require.Nil(t, extraHeaders)
+	require.Errorf(t, err, invalidExtraHeadersError)
+}
+
+func TestLambdaPromtail_TestParseHeadersNoneProvided(t *testing.T) {
+	extraLabels, err := parseExtraHeaders("")
+	require.Len(t, extraLabels, 0)
+	require.Nil(t, err)
+}
```

Comment:
lets simplify a bit and make this a table driven test

Reply:
I am not really sure how a table driven test simplifies my test, but I am also not too experience with this kind of testing. Maybe you could give me a hint and what you had in mind here?

Reply:
rather than declaring 4 new tests, one for each of these success or failure scenarios, we would have one test with an array of input/expected output, see here https://go.dev/wiki/TableDrivenTests